### PR TITLE
Fixing of AMQP and MQTT connection (issue#409)

### PR DESCRIPTION
**Pull request by Madhu-NEC**
Updating files for [Issue#409](https://github.com/telefonicaid/iotagent-json/issues/409)

### CODE REVIEW

**Comment by Madhu-NEC** [comment]
Hi @fgalan ,
I have committed the fix for [issue#409](https://github.com/telefonicaid/iotagent-json/issues/409) and all the test cases were passing on local setup by using below mentioned steps:
1. git clone https://github.com/Madhu-NEC/iotagent-json
2. cd iotagent-json
3. npm install
4. npm test

Please find below screenshot for reference:

test

But the result of all the test cases are showing as failed with travis.
Could you please let me know if there is any additional steps are required?

**Comment by fgalan** [comment]
Sometimes we have a test case related to AMQP which fails from time to time (a glitch, probably due to a timeout).

However, in this case, the problem seems to be other because all the tests are failing:

```
  0 passing (1m)
  44 failing
```

I'll try to run your branch in my local to see how it goes. If it also works there we should investigate what is doing travis different from local so it fails but local not.

**Comment by fgalan** [comment]
Another interesting fact is that your local pass shows 99 passing + 3 pending (the same as the last sucessfull pass on master branc: https://travis-ci.org/telefonicaid/iotagent-json/jobs/578216701) while travis shows 0 passing + 44 failing.

Although maybe that's normal and npm test stops if too many tests are failing (not sure...).

**Comment by fgalan** [comment]
In seems that due to changes in travis (https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment) our configuration was not working. PR We have https://github.com/telefonicaid/iotagent-json/pull/435 (now merged into master) has supposely fixed it.

Thus, could you upgrade this PR branch merging master on it and see how it goes, please? Thanks!

**Comment by Madhu-NEC** [comment]
Hi @fgalan ,

I have upgraded the PR. Now all the test cases are executed successfully.

File: lib/bindings/AMQPBinding.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ function start(callback) {
             }
```
Comment (fgalan) [inline lib/bindings/AMQPBinding.js]: An entry in the CHANGES_NEXT_RELEASE file should be added. For instance: 

```
- Fix: avoid connections to AMQP when that transport is not included in configuration (#409)
```

Comment (Madhu-NEC) [reply lib/bindings/AMQPBinding.js]: > An entry in the CHANGES_NEXT_RELEASE file should be added. For instance:
> 
> ```
> - Fix: avoid connections to AMQP when that transport is not included in configuration (#409)
> ```
Hi @fgalan ,
I have added an entry to CHANGES_NEXT_RELEASE but due to such modification resolve conflict issue is occurring as "This branch has conflicts that must be resolved".

Comment (fgalan) [reply lib/bindings/AMQPBinding.js]: I think the conflict is easy to solve.

1. Take the version now in master
2. Add to it the entry: `" Fix: Avoid connections to AMQP and MQTT when these transports are not included in configuration (#409)"` (note the issue number: when a change is related wih an issue number it is a good idea to include it in CHANGES_NEXT_RELEASE also)

Comment (fgalan) [reply lib/bindings/AMQPBinding.js]: Github provides a nice GUI to solve small conflicts like that. Do you have the following button at the bottom of the PR? Click "Resolve conflicts"

imagen

**Pull request merged**